core/arch/arm/plat-sam/matrix.h

```diff
@@ -0,0 +1,44 @@
+/* ----------------------------------------------------------------------------
+ *         ATMEL Microcontroller Software Support
+ * ----------------------------------------------------------------------------
+ * Copyright (c) 2013, Atmel Corporation
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * - Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the disclaimer below.
+ *
+ * Atmel's name may not be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * DISCLAIMER: THIS SOFTWARE IS PROVIDED BY ATMEL "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT ARE
+ * DISCLAIMED. IN NO EVENT SHALL ATMEL BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef MATRIX_H
+#define MATRIX_H
+
+extern void matrix_write_protect_enable(unsigned int matrix_base);
+extern void matrix_write_protect_disable(unsigned int matrix_base);
+extern void matrix_configure_slave_security(unsigned int matrix_base,
+				unsigned int slave,
+				unsigned int srtop_setting,
+				unsigned int srsplit_setting,
+				unsigned int ssr_setting);
+extern int matrix_configure_peri_security(unsigned int *peri_id_array,
+					unsigned int size);
+
+extern vaddr_t matrix32_base(void);
+extern vaddr_t matrix64_base(void);
```

Comment:
matrix32/64_base() is being used in both matrix.c and main.c, hence it is included in the header.

I have fixed all other review comments in the updated pull request. Thanks for reviewing.